src/compiler/checker.ts

```diff
@@ -13027,6 +13027,10 @@ namespace ts {
             if (propName !== undefined) {
                 const prop = getPropertyOfType(objectType, propName);
                 if (prop) {
+                    if (accessNode && prop?.isDeprecated) {

```

Comment:
This is just after `if (prop)` so the `?` isn't needed:

```suggestion
                    if (accessNode && prop.isDeprecated) {
```